### PR TITLE
Added API for managing certification target cloud services

### PR DESCRIPTION
**Pull request by oxisto**
Added API for managing certification target cloud services:

- [x] Registration of new services
- [x] Listing all services
- [x] Retrieve a specific service
- [x] Update a specific service
- [x] Remove a service
- [x] Updating target cloud service metric configuration
- [x] Database persistance 
- [x] Added adequate tests

Closes https://github.com/clouditor/clouditor/issues/387

This PR *intentionally* does *not* combine the metric configurations with the assessment yet, as we need to untangle certain things in the Rego evaluation first.

### CODE REVIEW

File: cli/session.go [hunk 1]
```diff
@@ -261,7 +269,8 @@ func getTools(toComplete string) []string {
 	return tools
 }
 
-func getMetrics(toComplete string) []string {
+// TODO(oxisto): This could be an interesting use case for 1.18 Go generics
+func getMetrics(_ string) []string {
```
Comment (garuppel) [inline cli/session.go:273]:
```suggestion
func getMetrics() []string {
```

File: cli/session.go [hunk 2]
```diff
@@ -287,3 +296,30 @@ func getMetrics(toComplete string) []string {
 
 	return metrics
 }
+
+func getCloudServices(_ string) []string {
```
Comment (garuppel) [inline cli/session.go:300]:
```suggestion
func getCloudServices() []string {
```

File: cli/session.go [hunk 3]
```diff
@@ -218,23 +218,31 @@ func DefaultArgsShellComp(cmd *cobra.Command, args []string, toComplete string)
...
-func getTools(toComplete string) []string {
+func ValidArgsGetCloudServices(_ *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+	if len(args) != 0 {
+		return nil, cobra.ShellCompDirectiveNoFileComp
+	}
+
+	return getCloudServices(toComplete), cobra.ShellCompDirectiveNoFileComp
+}
+
+func getTools(_ string) []string {
```
Comment (garuppel) [inline cli/session.go:245]:
```suggestion
func getTools() []string {
```

File: cli/session.go (hunk 1 above)
Comment (oxisto) [reply cli/session.go:273]: This is intentional (future proof). The `toComplete` holds the typed input, we might want to include this in the request as a filter.

File: cli/session.go (hunk 2 above)
Comment (oxisto) [reply cli/session.go:300]: This is intentional (future proof). The `toComplete` holds the typed input, we might want to include this in the request as a filter.

File: cli/session.go (hunk 3 above)
Comment (oxisto) [reply cli/session.go:245]: This is intentional (future proof). The `toComplete` holds the typed input, we might want to include this in the request as a filter.

File: service/auth/auth.go [hunk 4]
```diff
@@ -192,8 +192,6 @@ func StartDedicatedAuthServer(address string) (sock net.Listener, server *grpc.S
 		return nil, nil, fmt.Errorf("could not listen: %w", err)
 	}
 
-	err = persistence.InitDB(true, "", 0)
-
 	if err != nil {
```
Comment (garuppel) [inline service/auth/auth.go:195]: Delete

Comment (garuppel) [inline service/auth/auth.go:195]:
```suggestion
```

File: service/auth/auth.go [hunk 5]
```diff
@@ -192,8 +192,6 @@ func StartDedicatedAuthServer(address string) (sock net.Listener, server *grpc.S
...
 		return nil, nil, fmt.Errorf("could not initialize in-memory DB: %w", err)
 	}
```
Comment (garuppel) [inline service/auth/auth.go:197]:
```suggestion
```

File: cli/commands/cloud/cloud.go [hunk 6]
```diff
@@ -0,0 +1,285 @@
...
+
+	_ = cmd.RegisterFlagCompletionFunc("id", cli.ValidArgsGetCloudServices)
+	_ = cmd.RegisterFlagCompletionFunc("name", cli.DefaultArgsShellComp)
+	_ = cmd.RegisterFlagCompletionFunc("description", cli.DefaultArgsShellComp)
+
+	return cmd
+}
+
+// NewGetMetricConfigurationCommand returns a cobra command for the `get-metric-configuration` subcommand
+// TODO(oxisto): Can we have something like cl cloud get <id> metric-configuration <id>?
```
Comment (garuppel) [inline cli/commands/cloud/cloud.go:227]: I don´t know, but we can also use flags for this.

**Review by garuppel — COMMENTED** [review]
Finished :smile:

**Pull request merged**